### PR TITLE
Autograding for rest of cpp-02

### PR DESCRIPTION
**Pull request by abolinsky**

### CODE REVIEW

File: 02_object_oriented/01_challenge/main.cpp [hunk 1]
```diff
@@ -0,0 +1 @@
+// TODO
```
Comment (dancer-yuma) [inline 02_object_oriented/01_challenge/main.cpp:1]: The test failed with the solution in another PR. Am I missing anything?

```sh
yuma.sumi@YA22080179 ~/projects/cpp-02-en/02_object_oriented/01_challenge % ./autograder/run             
-- Configuring done (0.2s)
-- Generating done (0.5s)
-- Build files have been written to: /Users/yuma.sumi/projects/cpp-02-en/02_object_oriented/01_challenge/autograder/build
[ 16%] Built target gtest
[ 33%] Built target gtest_main
[ 41%] Building CXX object CMakeFiles/Tests.dir/Users/yuma.sumi/projects/cpp-02-en/02_object_oriented/01_challenge/car.cpp.o
[ 50%] Building CXX object CMakeFiles/Tests.dir/Users/yuma.sumi/projects/cpp-02-en/02_object_oriented/01_challenge/logger.cpp.o
[ 58%] Building CXX object CMakeFiles/Tests.dir/Users/yuma.sumi/projects/cpp-02-en/02_object_oriented/01_challenge/tests/tests.cpp.o
[ 66%] Linking CXX executable Tests
ld: warning: ignoring duplicate libraries: 'lib/libgtest.a'
[ 66%] Built target Tests
[ 83%] Built target gmock
[100%] Built target gmock_main
Running main() from /Users/yuma.sumi/projects/cpp-02-en/autograder/googletest/googletest/src/gtest_main.cc
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from Test
[ RUN      ] Test.Logger
Changing price from 40000.000000 to 35999.000000
/Users/yuma.sumi/projects/cpp-02-en/02_object_oriented/01_challenge/tests/tests.cpp:20: Failure
Expected equality of these values:
  Logger::was_created
    Which is: false
  true

[  FAILED  ] Test.Logger (0 ms)
[----------] 1 test from Test (0 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test suite ran. (0 ms total)
[  PASSED  ] 0 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] Test.Logger

 1 FAILED TEST
```

**Review by dancer-yuma — COMMENTED** [review]
@abolinsky 
I didn't dive deep into code, but checked the followings:
- run tests without solutions (should fail)
- run tests with solutions (should pass)

One challenge couldn't pass a test. It could be due to a solution or maybe tests, can you take a look?

File: 02_object_oriented/01_challenge/main.cpp (hunk 1 above)
Comment (abolinsky) [reply 02_object_oriented/01_challenge/main.cpp:1]: Great catch. I don't know how this sneaked through. Looking into it now.

**Review by dancer-yuma — APPROVED** [review]
@abolinsky 
LGTM!

**Review by dancer-yuma — APPROVED** [review]
@abolinsky 
Re-approving

Note: Challenge 5 was removed due to the content change

**Pull request merged**